src/python/pants/engine/target.py

```diff
@@ -1902,6 +1925,36 @@ def to_unparsed_address_inputs(self) -> UnparsedAddressInputs:
         return UnparsedAddressInputs(self.value or (), owning_address=self.address)
 
 
+class BinaryTargetDependencies(SpecialCasedDependencies):
+    """A base class for dependencies of a target that represent a binary artifact.
+
+    A "binary target" is one that:
+
+    A) Does not own any source code itself.
+    B) Provides dependencies to its dependees - if at all - from built artifacts,
+       rather than from source. These artifacts are built from source code in the binary target's
+       dependencies.
+
+    These dependencies are special-cased because a dependee on a binary target does not want
+    to transitively depend on that target's own dependencies, since those dependencies should be
+    provided - if at all - by an artifact.
+
+    Note: this is for the `dependencies` field *of* binary targets, not dependencies *on*
+      binary targets. The latter can be regular dependencies. This field indicates that
+      transitive dependency following "stops" at a binary target (except for in specific
+      use-cases like graph introspection).
+    """
+
+    alias = Dependencies.alias
+    help = (
+        Dependencies.help + "\n\nYou can also exclude a target or file from the entire graph "
+        "by prefixing it with `!!`."
+    )
```

Comment:
Should probably be applied to TestTargets too

Reply:
Done, via a shared base class (which we probably wanted anyway, since right now it looks like `SpecialCasedDependencies` subclasses `Dependencies` by naming convention, but it actually doesn't. Now at least there is `DependenciesBase`).